src/main/kotlin/io/github/kryszak/gwatlin/http/serializers/NumberSerializer.kt

```diff
@@ -0,0 +1,20 @@
+package io.github.kryszak.gwatlin.http.serializers
+
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.json.JsonContentPolymorphicSerializer
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonPrimitive
+
+/**
+ * Selects either [Int] or [Double] based on the JsonElement's content.
+ */
+class NumberSerializer : JsonContentPolymorphicSerializer<Number>(Number::class) {
+    override fun selectDeserializer(element: JsonElement): DeserializationStrategy<Number> {
+        require(element is JsonPrimitive) { "Expected a json primitive, got  $element" }
+        val content = element.content
+        require(!element.isString) { "String element not supported: '$content'" }
+        return if (content.toIntOrNull() != null) Int.serializer()
```

Comment:
I'd "kotlinize" this part like:
```suggestion
        return content.toIntOrNull()?.let { Int.serializer() } ?: Double.serializer()
```
and even better extract boolean function to check if content is and integer - for readability

Reply:
Something like:
```
fun isInteger(content: String) = content.toIntOrNull()?.let { true } ?: false
```

Reply:
I moved `isInteger` and `requireContent` to their own methods. The `selectDeserializer` implementation is clearer now, for sure. It's different now, thoughts?

Reply:
A lot clearer now 👌